### PR TITLE
[facts] [hardware] add cpu utilization to facts

### PR DESCRIPTION
**Pull request by KB-perByte**
##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```
switch# show processes cpu | include CPU
CPU utilization for five seconds: 1%/0%; one minute: 13%; five minutes: 11%
```

```paste below
ok: [nxos_int_test1] => changed=false 
  ansible_facts:
    ansible_net_api: cliconf
    ansible_net_cpu_utilization:
      core:
        five_minutes: 13
        five_seconds: 26
        five_seconds_interrupt: 0
        one_minute: 12
    ansible_net_filesystems:
    - 'bootflash:'
    ansible_net_gather_network_resources: []
    ansible_net_gather_subset:
    - default
    - hardware
    ansible_net_hostname: switch
    ansible_net_image: bootflash:///nxos64-cs.10.3.1.F.bin
    ansible_net_license_hostid: 9J642HUPL8Y
    ansible_net_memfree_mb: 2273.12109375
    ansible_net_memtotal_mb: 7946.30859375
    ansible_net_model: Nexus9000 C9300v Chassis
    ansible_net_platform: N9K-C9300v
    ansible_net_python_version: 3.10.12
    ansible_net_serialnum: 94JC5X73ZLT
    ansible_net_system: nxos
    ansible_net_version: 10.3(1)
    ansible_network_resources: {}
  invocation:
    module_args:
      available_network_resources: false
      gather_network_resources: null
      gather_subset:
      - hardware
```

### CODE REVIEW

File: plugins/module_utils/network/nxos/facts/legacy/base.py [hunk 1]
```diff
@@ -162,6 +162,47 @@ def populate(self):
...
                 self.facts["memfree_mb"] = self.parse_memfree_mb(data)
 
+        data = None
+        data = self.run("show processes cpu | include CPU")
+
+        if data:
+            self.facts["cpu_utilization"] = self.parse_cpu_utilization(data)
+
+    def parse_cpu_utilization(self, data):
+        facts = dict()
```
Comment (NilashishC) [inline plugins/module_utils/network/nxos/facts/legacy/base.py:172]: Just one of my pet peeves. :)
```suggestion
        facts = {}
```

File: plugins/module_utils/network/nxos/facts/legacy/base.py [hunk 2]
```diff
@@ -162,6 +162,47 @@ def populate(self):
                 self.facts["memtotal_mb"] = self.parse_memtotal_mb(data)
                 self.facts["memfree_mb"] = self.parse_memfree_mb(data)
 
+        data = None
+        data = self.run("show processes cpu | include CPU")
```
Comment (NilashishC) [inline plugins/module_utils/network/nxos/facts/legacy/base.py:166]: Can we just the `| json` or `| json-pretty` filter here? That would hopefully help reduce the amount of regexes needed.

**Comment by KB-perByte** [comment]
@NilashishC I am yet to add unit tests to this PR, I'll update it ^_^
thanks

File: plugins/module_utils/network/nxos/facts/legacy/base.py (hunk 2 above)
Comment (KB-perByte) [reply plugins/module_utils/network/nxos/facts/legacy/base.py:166]: I used json pretty it gave no output, checking it again!

**Review by NilashishC — APPROVED** [review]

**Pull request merged**